framework/py/flwr/cli/run/run.py

```diff
@@ -101,11 +102,30 @@ def run(
     try:
         if suppress_output:
             redirect_output(captured_output)
+
+        app_id_str = str(app) if app is not None else ""
+        is_remote_app: bool = False
+
+        if app_id_str.startswith("@"):
+            m = re.match(r"^@(?P<user>[^/]+)/(?P<app>[^/]+)$", app_id_str)
+            if not m:
+                raise typer.BadParameter(
+                    "Invalid remote app ID. Expected format: '@user_name/app_name'."
+                )
+            is_remote_app = True
```

Comment:
Is `app_id` used when it's not a remote app? If not, I think it would be nice to make `app_id` an `Optional[str]` instead of introducing a new boolean variable `is_remote_app`. Wdyt? For example:

```suggestion
        app_id = None
        if (app_str := str(app)).startswith("@"):
            if not re.match(r"^@(?P<user>[^/]+)/(?P<app>[^/]+)$", app_str):
                raise typer.BadParameter(
                    "Invalid remote app ID. Expected format: '@user_name/app_name'."
                )
            app_id = app_str
```